### PR TITLE
New Bypasses and Minor Fixes

### LINKED ISSUE #488: work.ink not being bypassed
~~We already know that linkvertise is not being bypassed, and we are working on it. Please be patient while we found a fix for it.~~
Edit 5:50PM at 17/05/22 (BRT): Pull Request [498](https://github.com/FastForwardTeam/FastForward/pull/498) fixed Linkvertise bypass. Work.ink still not bypassed.

### PR DESCRIPTION
**Pull request by ghost**
Fixes (Links to issues fixed by this PR): #488

Description:
Added 3 new bypasses (work.ink finally), and fixed some minor issues with the backend scripts. Want to fix infobox.js so that it uses local icon instead of remote file from Github.

Test links:
https://work.ink/4dw/lg3vkbak
[https://mega-guy.com](https://mega-guy.com/s?RaL)
[https://ofpacksmega.com](https://ofpacksmega.com/s?eyQ)

Checklist:

- [x] I made sure there are no unnecessary changes in the code*
- [x] Tested on Chromium- Browser OS
- [ ] Tested on Firefox

### CODE REVIEW

File: src/bypasses/lnk.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  constructor() {
+    super();
+    // custom bypass required bases can be set here
+  }
+
+  execute() {
+    if (!window.location.href.includes('/go/')) {
+      window.location.href = window.location.href.replace('lnk.parts', 'lnk.parts/go');
+    }
+    setTimeout(500);
```
Comment (lem6ns) [inline src/bypasses/lnk.js]: setTimeout is not synchronous (no idea if it is in this context)

**Review by lem6ns — CHANGES_REQUESTED** [review]

File: src/bypasses/lnk.js (hunk 1 above)
Comment (ghost) [reply src/bypasses/lnk.js]: You're right, it does nothing anyways.

File: src/bypasses/workclick.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  async execute() {
+    const uuid = new URLSearchParams(window.location.search).get('t');
+    this.helpers.insertInfoBox("You will need to wait for 10 seconds as the site is loading the destination URL. Please be patient.");
+    fetch(`https://redirect-api.work.ink/externalPopups/${uuid}/pageOpened`);
+    await new Promise(r => setTimeout(r, 11 * 1000));
+    const { destination } = await fetch(`https://redirect-api.work.ink/externalPopups/${uuid}/destination`).then(r => r.json());
+    const url = new URL(destination);
+    if (url.searchParams.has('duf')) {
+      const finalUrl = atob(url.searchParams.get('duf').split('').reverse().join(''));
```
Comment (NotAProton) [inline src/bypasses/workclick.js]:
```suggestion
      const finalUrl = window.atob(url.searchParams.get('duf').split('').reverse().join(''));
```

File: src/bypasses/workink.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                }
+                break;
+            case "s_monetization":
+                if (payload.type !== "browserExtension") break;
+                ws.send(payloads.browserExtension["2"](payload.payload.token))
+                break;
+            case "s_link_destination":
+                const url = new URL(payload.url);
+                if (url.searchParams.has("duf")) {
+                    return this.helpers.safelyNavigate(atob(url.searchParams.get("duf").split("").reverse().join("")))
```
Comment (NotAProton) [inline src/bypasses/workink.js]:
```suggestion
                    return this.helpers.safelyNavigate(window.atob(url.searchParams.get("duf").split("").reverse().join("")))
```

File: src/helpers/infobox.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ export function insertInfoBox(text) {
   const div = document.createElement('div');
   div.style =
     'margin-left:20px; margin-bottom: 20px;background:#eee;border-radius:10px;padding:20px;box-shadow:#111 0px 5px 40px;cursor:pointer';
-  div.innerHTML = `<img src="${window.x8675309bp}icon/48.png" style="width:24px;height:24px;margin-right:8px"><span style="display:inline"></span>`;
+  div.innerHTML = `<img src="https://raw.githubusercontent.com/FastForwardTeam/FastForward/main/src/icon/48.png" style="width:24px;height:24px;margin-right:8px"><span style="display:inline"></span>`;
```
Comment (NotAProton) [inline src/helpers/infobox.js]: use the internal one packed with the extension

File: src/js/content_script.js [hunk 5]
```diff
@@ -56,7 +56,7 @@ document.addEventListener('ff53054c0e13_crowdQuery', async function (event) {
   if (options.optionCrowdBypass === false) {
     const src = brws.runtime.getURL('helpers/infobox.js');
     const insertInfoBox = await import(src);
-    insertInfoBox(brws.i18n.getMessage('crowdDisabled'));
+    this.helpers.insertInfoBox(brws.i18n.getMessage('crowdDisabled'));
```
Comment (NotAProton) [inline src/js/content_script.js]: I don't think you need that

**Review by NotAProton — CHANGES_REQUESTED** [review]
vscode thinks atob is deprecated use window.atob pls

**Comment by lem6ns** [comment]
resolves #488

File: src/helpers/infobox.js (hunk 4 above)
Comment (ghost) [reply src/helpers/infobox.js]: How do I do this? The extension ID function isn't here and the old one depended on {window.x8675309bp}, which we know doesn't work anymore.

**Review by lem6ns — APPROVED** [review]

**Comment by lostdusty** [comment]
ready?

File: src/helpers/infobox.js (hunk 4 above)
Comment (lostdusty) [reply src/helpers/infobox.js]: Can't file://${pwd} be used?

File: src/bypasses/workclick.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+export default class Workclick extends BypassDefinition {
+  constructor() {
+    super();
+    // custom bypass required bases can be set here
+  }
+
+  async execute() {
+    const uuid = new URLSearchParams(window.location.search).get('t');
+    this.helpers.insertInfoBox("You will need to wait for 10 seconds as the site is loading the destination URL. Please be patient.");
```
Comment (NotAProton) [inline src/bypasses/workclick.js:11]: What about i18n?

**Pull request merged**